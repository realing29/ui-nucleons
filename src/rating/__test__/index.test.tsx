import React from 'react';
import { Rating } from '..';
import { mount } from 'enzyme';

describe('<Rating />', () => {
  it('renders correctly without params', () => {
    const wrapper = mount(
      <Rating value={3.4} className='test-class' />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
