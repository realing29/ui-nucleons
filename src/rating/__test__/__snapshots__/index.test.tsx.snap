// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> should handle size prop missing 1`] = `
<div>
  <div
    class="root"
    data-rating="3.4"
    data-testid="rating"
  >
    <div
      class="star size-s"
    >
      <svg
        class="svg full"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
    </div>
    <div
      class="star size-s"
    >
      <svg
        class="svg full"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
    </div>
    <div
      class="star size-s"
    >
      <svg
        class="svg full"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
    </div>
    <div
      class="star size-s"
    >
      <svg
        class="svg"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
      <svg
        class="svg half"
        height="24"
        preserveAspectRatio="xMinYMin slice"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
    </div>
    <div
      class="star size-s"
    >
      <svg
        class="svg"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`<Rating /> should renders correctly 1`] = `
<div>
  <div
    class="root test-class"
    data-rating="3.4"
    data-testid="rating"
  >
    <div
      class="star size-m"
    >
      <svg
        class="svg full"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
    </div>
    <div
      class="star size-m"
    >
      <svg
        class="svg full"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
    </div>
    <div
      class="star size-m"
    >
      <svg
        class="svg full"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
    </div>
    <div
      class="star size-m"
    >
      <svg
        class="svg"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
      <svg
        class="svg half"
        height="24"
        preserveAspectRatio="xMinYMin slice"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
    </div>
    <div
      class="star size-m"
    >
      <svg
        class="svg"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.304 2.318a.75.75 0 0 1 1.392 0l2.567 6.393 6.874.466a.75.75 0 0 1 .43 1.324l-5.287 4.417 1.68 6.681a.75.75 0 0 1-1.125.818L12 18.754l-5.835 3.663A.75.75 0 0 1 5.04 21.6l1.681-6.68L1.433 10.5a.75.75 0 0 1 .43-1.324l6.874-.466 2.567-6.393Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
